src/x86_64/pvclock.c

```diff
@@ -34,5 +40,50 @@ void init_pvclock(heap h, struct pvclock_vcpu_time_info *vti)
 {
     assert(vti);
     vclock = vti;
+    pvclock_heap = h;
     register_platform_clock_now(closure(h, pvclock_now));
 }
+
+closure_function(0, 0, void, tsc_deadline_interrupt)
+{
+    /* debug here */
+}
+
+closure_function(0, 1, void, tsc_deadline_timer,
```

Comment:
Minor quibble, but Is it possible to abstract the vclock stuff out from here so that pvclock_now() and tsc_deadline_timer() don't have to duplicate this logic?

Reply:
Aside from the version check which is a single conditional, these two are actually the reverse of each other (TSC interval -> timestamp for pvclock_now and the reverse for the timer).